pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractReplicator.java

```diff
@@ -211,35 +265,67 @@ protected synchronized CompletableFuture<Void> closeProducerAsync() {
         });
     }
 
-
-    public CompletableFuture<Void> disconnect() {
-        return disconnect(false);
+    public synchronized CompletableFuture<Void> terminate() {
+        if (producer == null) {
```

Comment:
Do we need to add `tryChangeStatusToTerminating();` before this check?

Reply:
Added. And I moved `synchronized` after changing the status to `Terminating` successfully.